aws/data_source_aws_wafv2_ip_set_test.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccDataSourceAwsWafv2IPSet_Basic(t *testing.T) {
+	name := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_wafv2_ip_set.test"
+	datasourceName := "data.aws_wafv2_ip_set.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccDataSourceAwsWafv2IPSet_NonExistent(name),
+				ExpectError: regexp.MustCompile(`WAFv2 IPSet not found`),
+			},
+			{
+				Config: testAccDataSourceAwsWafv2IPSet_Name(name),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair(datasourceName, "addresses", resourceName, "addresses"),
```

Comment:
we should also use the ARN testing check function in this step as well

Reply:
The ARN is tested against the ARN of the resource, does this not guarantee it's correct? 🤔

Reply:
ah yes good catch! for ARN checking `resource.TestCheckResourceAttrPair` or `testAccMatchResourceAttrRegionalARN` are valid so either one will suffice